activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -476,6 +476,8 @@ def get_period_and_ensure_valid_local_time(period)
       end
 
       def transfer_time_values_to_utc_constructor(time)
+        # avoid creating another Time object if possible
+        return time if time.utc? && time.is_a?(Time)
```

Comment:
This method is currently aways returning a standard Ruby `Time` object, regardless of what type of duck-typed time class you pass into it. This PR would change that because `is_a?` includes inheritance.


`transfer_time_values_to_utc_constructor(Time.zone.now)` would now return an instance of `ActiveSupport::TimeWithZone` instead of `Time` like it currently does.

```ruby
irb(main):011:0> Time.zone.now.class
=> ActiveSupport::TimeWithZone
irb(main):012:0> Time.zone.now.is_a?(Time)
=> true
irb(main):013:0> Time.zone.now.instance_of?(Time)
=> false
```

I think what you really want is the following:

```ruby
return time if time.instance_of?(Time) && time.utc?
```